In [1]:
import networkx as nx
import pandas as pd
import matplotlib.pyplot as plt

In [27]:
pip install --upgrade scipy networkx 

Note: you may need to restart the kernel to use updated packages.


In [28]:
nodes_heros = pd.read_csv('C:/Users/Chiara Sammarco/Desktop/Università/DATA SCIENCE - Magistrale/Esami da fare/ALGORITHMIC METHODS IN DATA MINING/HW 5/nodes.csv')
edges_heros = pd.read_csv('C:/Users/Chiara Sammarco/Desktop/Università/DATA SCIENCE - Magistrale/Esami da fare/ALGORITHMIC METHODS IN DATA MINING/HW 5/edges.csv')

In [29]:
edges_heros

hero     comic
0      24-HOUR MAN/EMMANUEL    AA2 35
1      3-D MAN/CHARLES CHAN     AVF 4
2      3-D MAN/CHARLES CHAN     AVF 5
3      3-D MAN/CHARLES CHAN     COC 1
4      3-D MAN/CHARLES CHAN    H2 251
...                     ...       ...
96099                 ZZZAX    H2 326
96100                 ZZZAX    H2 327
96101                 ZZZAX  M/CP 8/4
96102                 ZZZAX     PM 47
96103                 ZZZAX   WCA2 12

[96104 rows x 2 columns]

In [30]:
nodes_heros

node   type
0                   2001 10  comic
1                    2001 8  comic
2                    2001 9  comic
3      24-HOUR MAN/EMMANUEL   hero
4      3-D MAN/CHARLES CHAN   hero
...                     ...    ...
19085           ZOTA, CARLO   hero
19086                  ZOTA   hero
19087                 ZURAS   hero
19088                  ZURI   hero
19089                 ZZZAX   hero

[19090 rows x 2 columns]

In [33]:
G = nx.Graph()
G = nx.from_pandas_edgelist(edges_heros, 'comic', 'hero')

In [34]:
nx.info(G)

C:\Users\Chiara Sammarco\AppData\Local\Temp\ipykernel_10308\937291259.py:1: DeprecationWarning: info is deprecated and will be removed in version 3.0.

  nx.info(G) # Se loro intendono con "graphs setup" solamente avere il grafo e non letteralmente il plt.show() allora penso


'Graph with 19090 nodes and 96104 edges'

We checked that the graph created using ONLY the edges dataframe has the right number of nodes and right number of edges (it does because nx.info() has output: 'Graph with 19090 nodes and 96104 edges' and the nodes dataframe, that we didn't use, has 19090 nodes, so it works).

In [2]:
# If you want to run the graph as plot:

# plt.clf()
# nx.draw(G, with_labels=True)
# plt.show()

# It takes a while so if it's not required I wouldn't run it.
# If it is required, then we have to fix it (make it bigger and put different colors depending on the type of node
# we have, either a 'comic' node or a 'hero' node)